content/redirect.js

```diff
@@ -2,18 +2,68 @@ const fs = require("fs");
 const path = require("path");
 
 const { resolveFundamental } = require("../libs/fundamental-redirects");
-const { CONTENT_ROOT, VALID_LOCALES } = require("./constants");
+const { decodePath, slugToFolder } = require("../libs/slug-utils");
+const {
+  CONTENT_ROOT,
+  CONTENT_TRANSLATED_ROOT,
+  VALID_LOCALES,
+} = require("./constants");
+
+const FORBIDDEN_URL_SYMBOLS = ["\n", "\t"];
+
+let ARCHIVED_URLS;
+
+function checkURLInvalidSymbols(url) {
+  for (const character of FORBIDDEN_URL_SYMBOLS) {
+    if (url.includes(character)) {
+      throw new Error(`URL contains invalid character '${character}'`);
+    }
+  }
+}
+
+function documentExists(url) {
+  // Let's keep vanity urls to /en-US/
+  if (url === "/en-US/") {
+    return url;
+  }
+  const [bareURL] = url.split("#", 2);
+
+  if (!ARCHIVED_URLS) {
+    ARCHIVED_URLS = new Set(
+      fs
+        .readFileSync(path.join(__dirname, "archived.txt"), "utf-8")
+        .split("\n")
```

Comment:
```suggestion
        .split("\n")
        .filter(line => !(line.startsWith('#') || !line.trim()))
```
perhaps?

Reply:
I guess that out to be 
```suggestion
        .split("\n")
        .filter(line => !line.startsWith('#') || line.trim())
```
neater, right?